linkerd/core/src/main/scala/io/buoyant/linkerd/RequestActiveTracer.scala

```diff
@@ -0,0 +1,263 @@
+package io.buoyant.linkerd
+
+import com.twitter.finagle._
+import com.twitter.finagle.client.Transporter.EndpointAddr
+import com.twitter.finagle.http._
+import com.twitter.finagle.http.Fields.MaxForwards
+import com.twitter.finagle.http.Status
+import com.twitter.finagle.http.util.StringUtil
+import com.twitter.finagle.naming.buoyant.DstBindingFactory
+import com.twitter.util._
+import io.buoyant.namer.DelegateTree._
+import io.buoyant.namer.{DelegateTree, Delegator}
+import io.buoyant.router.RouterLabel
+import io.buoyant.router.RoutingFactory
+import io.buoyant.router.RoutingFactory.BaseDtab
+import io.buoyant.router.context.{DstBoundCtx, DstPathCtx}
+
+/**
+ * Intercepts Http TRACE requests with a Max-Forwards and l5d-add-context header.
+ * Returns identification, delegation, and service address information
+ * for a given router as well as any other TRACE responses received from downstream
+ * services.
+ *
+ * A router that receives a tracer request with an l5d-max-depth
+ * greater than 0 decrements the l5d-max-depth header by 1 and
+ * forwards the tracer request to a downstream service that may perform
+ * additional processing of the request.
+ *
+ * @param endpoint the endpoint address of a downstream service identified by a router
+ * @param namers namers used to evaluate a service name
+ * @param dtab base dtab used by the router
+ * @param routerLabel name of router
+ */
+class RequestActiveTracer(
+  endpoint: EndpointAddr,
+  namers: DstBindingFactory.Namer,
+  dtab: BaseDtab,
+  routerLabel: String
+) extends SimpleFilter[Request, Response] {
+
+  private[this] case class PathNode(
+    path: String,
+    dentry: String,
+    dTreeNode: DelegateTree[_] = DelegateTree.Empty(Path.empty, Dentry.nop)
+  ) {
+    override def toString: String = {
+      dTreeNode match {
+        case DelegateTree.Delegate(_, d, _) if d == Dentry.nop => s"$path"
+        case _ => s"$path ($dentry)"
+      }
+
+    }
+  }
+
+  private[this] val AddRouterContextHeader = "l5d-add-context"
+
+  private def formatRouterContext(
+    prependText: String,
+    serviceName: String,
+    clientName: String,
+    selectedAddress: String,
+    addresses: Option[Set[String]],
+    dtabResolution: List[String],
+    elapsedTimestamp: String
+  ) = {
+    Seq(
+      prependText,
+      s"""
+      |--- Router: $routerLabel ---
+      |request duration: $elapsedTimestamp ms
+      |service name: $serviceName
+      |client name: $clientName
+      |addresses: [${addresses.getOrElse(Set.empty).mkString(", ")}]
+      |selected address: $selectedAddress
+      |dtab resolution:
+      |${dtabResolution.map("  " + _).mkString("\n")}
+    """.stripMargin
+    ).mkString("\n").trim.concat("\n")
+  }
+
+  /**
+   * Returns a list of [[PathNode]] that represents a delegate tree path that identifies how a
```

Comment:
Scaladoc links using `[[` and `]]` have a tendency to break artifact publishing.  I don't know why, but I've had to remove them in the past.